Dockerfile

```diff
@@ -164,16 +166,24 @@ COPY --from=builder \
 COPY --from=builder /build/vcftools/built/bin/    /usr/local/bin/
 COPY --from=builder /build/vcftools/built/share/  /usr/local/share/
 
+ARG TARGETPLATFORM
+
+# use /env to persist variable across RUNs
+RUN if [ "$TARGETPLATFORM" = "linux/amd64" ]; then echo "export NEXTCLADE_PLATFORM=x86_64-unknown-linux-gnu" >> /env; fi
+RUN if [ "$TARGETPLATFORM" = "linux/arm64" ]; then echo "export NEXTCLADE_PLATFORM=aarch64-unknown-linux-gnu" >> /env; fi
+
 # Add Nextalign v2
-RUN curl -fsSL https://github.com/nextstrain/nextclade/releases/latest/download/nextalign-x86_64-unknown-linux-gnu \
+RUN . /env; curl -fsSL https://github.com/nextstrain/nextclade/releases/latest/download/nextalign-$NEXTCLADE_PLATFORM \
```

Comment:
There's also `TARGETARCH` [available](https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope), and `ARG`s are available in the environment of a `RUN` (which is [not subject to variable substitution](https://docs.docker.com/engine/reference/builder/#environment-replacement) at the Dockerfile level), so an alternative to this unusual sort of roundabout `./env` thing is something like:

```Dockerfile
ARG TARGETARCH
RUN curl …/download/nextalign-$(case $TARGETARCH in arm64) echo aarch64;; amd64) echo x86_64;; esac)-unknown-linux-gnu
```

The mapping of Docker-style archs to standard/Rust-style archs is an unfortunate necessity.

Reply:
I implemented another alternative: 70443f60fa6d33f9e16780db6ca5dbbd491f3a1b

This puts the conditional logic in a script, which is then invoked inline to construct the URL.

Also, I chose `TARGETPLATFORM` over `TARGETARCH` since I think it's nice to include the OS component in the mapping when it's included in the URLs as well. This also keeps the door open for other OS builds, e.g.

- `linux/amd64`-> `x86_64-unknown-linux-gnu`
- `linux/arm64` -> `aarch64-unknown-linux-gnu`
- `windows/amd64` -> `x86_64-pc-windows-gnu.exe`